core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/PojoTypes.java

```diff
@@ -29,7 +29,9 @@ public class PojoTypes implements JdbiConfig<PojoTypes> {
     public PojoTypes() {}
 
     private PojoTypes(PojoTypes other) {
-        factories.putAll(other.factories);
+        synchronized (other.factories) {
```

Comment:
e.g. PostgresTypes uses a ConcurrentHashMap, which is (to me) the preferred approach over adding synchronized blocks.